### PR TITLE
PAY-3826 : Fix to Duplicate Payment Issue in Liberata Pull API

### PR DESCRIPTION
**Pull request by AyanDuttaHMCTS**
### JIRA link (if applicable) ###

PAY-3826 : Fix to Duplicate Payment Issue in Liberata Pull API

### Change description ###

PAY-3826 : Fix to Duplicate Payment Issue in Liberata Pull API
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java [hunk 1]
```diff
@@ -180,7 +180,7 @@ public void makeAndRetrievePbaPaymentByFinrem() {
             .then()
             .statusCode(CREATED.value())
             .body("status", equalTo("Success"));
-
+        Thread.sleep(1000);
```
Comment (AyanDuttaHMCTS) [inline api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java:183]: Test cases to be refactored - Tech Backlog

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java [hunk 2]
```diff
@@ -202,7 +202,7 @@ public void makeAndRetrievePbaPaymentByUnspecService() {
...
```
Comment (AyanDuttaHMCTS) [inline api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java:205]: Test cases to be refactored - Tech Backlog

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java [hunk 3]
```diff
@@ -225,6 +225,7 @@ public void makeAndRetrievePbaPaymentByIACService() {
             .statusCode(CREATED.value())
             .body("status", equalTo("Success"));
 
+        Thread.sleep(1000);
```
Comment (AyanDuttaHMCTS) [inline api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java:228]: Test cases to be refactored - Tech Backlog

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java [hunk 4]
```diff
@@ -247,6 +248,7 @@ public void makeAndRetrievePbaPaymentByFPLService() {
...
```
Comment (AyanDuttaHMCTS) [inline api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java:251]: Test cases to be refactored - Tech Backlog

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PaymentRecordFunctionalTest.java [hunk 5]
```diff
@@ -71,6 +71,12 @@ public void createPaymentRecordAndValidateSearchResults() throws Exception {
             .then().created(paymentDto -> {
             assertNotNull(paymentDto.getReference());
 
+            try {
+                Thread.sleep(1000);
```
Comment (AyanDuttaHMCTS) [inline api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PaymentRecordFunctionalTest.java:75]: Test cases to be refactored - Tech Backlog

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PaymentsSearchFunctionalTest.java [hunk 6]
```diff
@@ -144,6 +144,8 @@ public void givenTwoPaymentsInPeriodWhensearchPaymentsWithStartDateEndDateThenSh
             assertEquals("payment status is properly set", "Initiated", paymentDto.getStatus());
         });
 
+        Thread.sleep(1000);
```
Comment (AyanDuttaHMCTS) [inline api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PaymentsSearchFunctionalTest.java:147]: Test cases to be refactored - Tech Backlog

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/TelephonyPaymentsTest.java [hunk 7]
```diff
@@ -97,6 +101,11 @@ public void retrieveASuccessfulTelephonyPaymentViaLookup() {
                 .when().updatePaymentStatus(referenceNumber, status)
                 .then().noContent();
 
+            try {
+                Thread.sleep(1000);
```
Comment (AyanDuttaHMCTS) [inline api/src/functionalTest/java/uk/gov/hmcts/payment/functional/TelephonyPaymentsTest.java:105]: Test cases to be refactored - Tech Backlog

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/TelephonyPaymentsTest.java [hunk 8]
```diff
@@ -138,6 +147,11 @@ public void retrieveAFailedTelephonyPaymentViaLookup() {
...
```
Comment (AyanDuttaHMCTS) [inline api/src/functionalTest/java/uk/gov/hmcts/payment/functional/TelephonyPaymentsTest.java:151]: Test cases to be refactored - Tech Backlog

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/TelephonyPaymentsTest.java [hunk 9]
```diff
@@ -179,6 +193,13 @@ public void retrieveAnErrorneousTelephonyPaymentViaLookup() {
                 .returnUrl("https://www.moneyclaims.service.gov.uk")
                 .when().updatePaymentStatus(referenceNumber, status)
                 .then().noContent();
+
+            try {
+                Thread.sleep(1000);
```
Comment (AyanDuttaHMCTS) [inline api/src/functionalTest/java/uk/gov/hmcts/payment/functional/TelephonyPaymentsTest.java:198]: Test cases to be refactored - Tech Backlog

File: api/src/main/java/uk/gov/hmcts/payment/api/controllers/PaymentController.java [hunk 10]
```diff
@@ -190,9 +190,23 @@ public PaymentDto retrievePayment(@PathVariable("reference") String paymentRefer
             .filter(p -> p.getReference().equals(reference)).findAny();
     }
 
-    private void populatePaymentDtos(final List<PaymentDto> paymentDtos, final PaymentFeeLink paymentFeeLink) {
+    private void populatePaymentDtos(final List<PaymentDto> paymentDtos, final PaymentFeeLink paymentFeeLink, Date fromDateTime, Date toDateTime) {
         //Adding this filter to exclude Exela payments if the bulk scan toggle feature is disabled.
         List<Payment> payments = getFilteredListBasedOnBulkScanToggleFeature(paymentFeeLink);
+
```
Comment (AyanDuttaHMCTS) [inline api/src/main/java/uk/gov/hmcts/payment/api/controllers/PaymentController.java:196]: Business Logics to be moved to Service Layer not in Controller

**Review by YasharRahvar — APPROVED** [review]

**Review by krishnanuthalapati — APPROVED** [review]

**Pull request merged**